File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        let i = 0;
        let j = numbers.length - 1;
        while(i<j) {
            const tempTotal = numbers[i] + numbers[j];
            if(tempTotal > target) {
                j--;
            } else if(tempTotal < target) {
                i++;
            } else if(tempTotal === target) {
                return [i+1,j+1];
            }
        }
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
// class Solution {
//     public int[] twoSum(int[] nums, int target) {
//         for(int i=0; i<nums.length; i++){
//             for(int j=0; j<nums.length; j++){
//                 if(nums[i] + nums[j] == target) {
//                     return new int[]{i,j};
//                 }
//             }
//         }
//         return new int[0];
//     }
// }

class Solution {
    public int[] twoSum(int[] nums, int target){
        HashMap<Integer, Integer> sumMap = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            int diff = target - nums[i];
            if(sumMap.containsKey(diff)) {
                return new int[]{sumMap.get(diff), i};
            }
            sumMap.put(nums[i], i);
        }

        return new int[] {};
    }
}
